FWCore/Framework/interface/EventSetup.h

```diff
@@ -142,7 +145,29 @@ class EventSetup
       //NOTE: the records are not owned
       std::map<eventsetup::EventSetupRecordKey, eventsetup::EventSetupRecord const *> recordMap_;
       eventsetup::EventSetupKnownRecordsSupplier const* knownRecords_;
-};
+  };
+
+  // Free functions to retrieve an object from the EventSetup.
+  // Will throw an exception if the record or  object are not found.
+
+  template <typename T, typename R = typename eventsetup::data_default_record_trait<typename T::value_type>::type>
```

Comment:
I've inverted the order of the template parameters, so that the record type can be deduced from the data type.